ldi-orchestrator/ldio-application/src/main/java/be/vlaanderen/informatievlaanderen/ldes/ldio/config/FlowAutoConfiguration.java

```diff
@@ -3,70 +3,82 @@
 
+import java.util.HashMap;
 import java.util.List;
+import java.util.Map;
 
 @Configuration
 @ComponentScan
 public class FlowAutoConfiguration {
+	private final OrchestratorConfig config;
+	private final ConfigurableApplicationContext configContext;
 
-	private final ApplicationContext applicationContext;
-
-	public FlowAutoConfiguration(ApplicationContext applicationContext) {
-		this.applicationContext = applicationContext;
+	public FlowAutoConfiguration(OrchestratorConfig config,
+			ConfigurableApplicationContext configContext) {
+		this.config = config;
+		this.configContext = configContext;
 	}
 
-	@Bean
-	@ConditionalOnMissingBean(LdiInput.class)
-	@DependsOn({ "componentExecutor" })
-	public LdiInput ldtoInput(OrchestratorConfig orchestratorConfig) {
-		return (LdiInput) getLdiComponent(orchestratorConfig.getInput().getName(),
-				orchestratorConfig.getInput().getConfig());
+	@PostConstruct
+	public void registerInputBeans() {
+		config.getPipelines().forEach(this::initialiseLdiInput);
 	}
 
-	@Bean("componentExecutor")
-	public ComponentExecutor componentExecutor(final OrchestratorConfig orchestratorConfig) {
-		List<LdiTransformer> ldiTransformers = orchestratorConfig.getTransformers()
+	public ComponentExecutor componentExecutor(final PipelineConfig pipelineConfig) {
+		List<LdiTransformer> ldiTransformers = pipelineConfig.getTransformers()
 				.stream()
-				.map(this::ldtoTransformer)
+				.map(this::getLdioTransformer)
 				.toList();
-		List<LdiOutput> ldiOutputs = orchestratorConfig.getOutputs()
+		List<LdiOutput> ldiOutputs = pipelineConfig.getOutputs()
 				.stream()
-				.map(this::ldtoOutput)
+				.map(this::getLdioOutput)
 				.toList();
 		return new ComponentExecutorImpl(ldiTransformers, ldiOutputs);
 	}
 
-	@Bean
-	public LdiAdapter ldiAdapter(final OrchestratorConfig orchestratorConfig) {
-		ComponentDefinition adapterDefinition = orchestratorConfig.getInput().getAdapter();
-
-		return (LdiAdapter) getLdiComponent(adapterDefinition.getName(), adapterDefinition.getConfig());
-	}
-
-	private LdiTransformer ldtoTransformer(ComponentDefinition componentDefinition) {
+	private LdiTransformer getLdioTransformer(ComponentDefinition componentDefinition) {
 		return (LdiTransformer) getLdiComponent(componentDefinition.getName(), componentDefinition.getConfig());
 	}
 
-	private LdiOutput ldtoOutput(ComponentDefinition componentDefinition) {
+	private LdiOutput getLdioOutput(ComponentDefinition componentDefinition) {
 		return (LdiOutput) getLdiComponent(componentDefinition.getName(), componentDefinition.getConfig());
 	}
 
+	public void initialiseLdiInput(PipelineConfig config) {
+		SingletonBeanRegistry beanRegistry = configContext.getBeanFactory();
+		LdioInputConfigurator configurator = (LdioInputConfigurator) configContext.getBean(
+				config.getInput().getName());
+		LdiAdapter adapter = (LdiAdapter) getLdiComponent(config.getInput().getAdapter().getName(),
+				config.getInput().getAdapter().getConfig());
```

Comment:
Law of Demeter: https://en.wikipedia.org/wiki/Law_of_Demeter
It's typically a code smell when you need to do object.getProperty().getProperty().getProperty()